crates/primitives/starknet/src/transaction/execution.rs

```diff
@@ -575,6 +601,82 @@ mod simulate_tx_offset {
         );
     }
 
-    // TODO: add test that check that each get_account_transaction_context impl correctly uses this
-    // offset
+    #[test]
+    fn transaction_version_trait_correctly_applies_simulate_tx_version_offset() {
+        let v0 = TransactionVersion(StarkFelt::from(0u8));
+        assert_eq!(v0.clone().apply_simulate_tx_version_offset(), apply_simulate_tx_version_offset(v0.clone(), true));
+
+        let v1 = TransactionVersion(StarkFelt::from(1u8));
+        assert_eq!(v1.clone().apply_simulate_tx_version_offset(), apply_simulate_tx_version_offset(v1.clone(), true));
+    }
+
+    #[test]
+    fn l1_handler_transaction_correctly_applies_simulate_tx_version_offset() {
+        let declare_tx = L1HandlerTransaction {
+            tx: Default::default(),
+            paid_fee_on_l1: Default::default(),
+            tx_hash: Default::default(),
+        };
+
+        let original_version = declare_tx.tx.version;
+
+        let queried_version = declare_tx.get_account_transaction_context(true).version;
+        eval_simulate_tx_version_offset(original_version.clone(), queried_version, true);
+
+        let non_queried_version = declare_tx.get_account_transaction_context(false).version;
+        eval_simulate_tx_version_offset(original_version.clone(), non_queried_version, false);
```

Comment:
This is hard to understand.
`eval_simulate_tx_version_offset` make it hard to know that is actually tested.

```rust
let queried_version = declare_tx.get_account_transaction_context(true).version;
let non_queried_version = declare_tx.get_account_transaction_context(false).version;
assert_eq!(queried_version, non_queried_version + SIMULATE_TX_VERSION_OFFSET)
```

is much more straightforward and give the same guarantee

Reply:
same for the other tests :)

Reply:
The above doesn't work, as we need to convert the version to a FieldElement first, before adding the offset. It got verbose and quite messy without impl a from trait, which I felt was overkill just for testing. I've worked on the function comment, making it clearer what happens. Let me know your thoughts on this.

Reply:
`eval_simulate_tx_version_offset(original_version_v0, queried_version, true);`  is not expressive at all. 
 Reading this I have no idea what is done, what is asserted against what.
 
 I would much prefer a `assert_eq!(queried_version, Felt252Wrapper(Felt252::from(non_queried_version) + SIMULATE_TX_VERSION_OFFSET))`.
 It's a bit verbose I agree, but at least I undersand what is being tested (even if the newtypes shenanigans are a bit obsucre. And we can use intermediate value for even more readability.
 
 
 When writing test we don't aim for the most well abstracted code. We want to be able to understand in a fraction of second what is being tested. And often the best way to have this expressiveness is by writing things in a much verbose way than what we would have done for the actual code

Reply:
Yeah fair, impl as requested